x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/InsertFieldExtraction.java

```diff
@@ -102,15 +100,18 @@ public PhysicalPlan apply(PhysicalPlan plan) {
 
     private static Set<Attribute> missingAttributes(PhysicalPlan p) {
         var missing = new LinkedHashSet<Attribute>();
-        var inputSet = p.inputSet();
+        var input = p.inputSet();
 
-        // TODO: We need to extract whatever fields are missing from the left hand side.
-        // skip the lookup join since the right side is always materialized and a projection
+        // For LOOKUP JOIN we only need field-extraction on left fields used to match, since the right side is always materialized
         if (p instanceof LookupJoinExec join) {
```

Comment:
As discussed, I think this can be simplified by just using `p.references` instead of having a special case for lookup joins, and otherwise doing something on `p.forEacExpression(...)`.